pkg/storage/chunk/client/ibmcloud/trusted_profile_authentication_provider.go

```diff
@@ -0,0 +1,159 @@
+package ibmcloud
+
+import (
+	"os"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/ibm-cos-sdk-go/aws/awserr"
+	"github.com/IBM/ibm-cos-sdk-go/aws/credentials"
+	"github.com/IBM/ibm-cos-sdk-go/aws/credentials/ibmiam/token"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/loki/pkg/util/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	trustedProfileProviderName = "TrustedProfileProviderNameIBM"
+	tokenType                  = "Bearer"
+)
+
+// TrustedProfileProvider Struct
```

Comment:
this needs further simplification:
`TrustedProfileProvider implements Provider interface from https://github.com/IBM/ibm-cos-sdk-go`